apps/charterafrica/src/lib/data/common/index.js

```diff
@@ -171,48 +171,41 @@ export async function processPageFellowships({ blocks }) {
   });
 }
 
-export async function processPageNews({ blocks }) {
-  // TODO(kilemensi): Pull data from CMS
-  blocks.push({
-    slug: "featured-post",
-    category: "News",
-    title: "News Story title goes here and spans over second line",
-    excerpt:
-      "Lorem ipsum dolor sit amet consectetur adipiscing elit tempus nibh cursus, urna porta sagittis non eget taciti nunc sed felis dui, praesent ullamcorper facilisi euismod ut in platea laoreet integer. Lorem ipsum dolor sit amet consectetur",
-    date: "2020-10-10 10:10:10",
-    author: "Author Name",
-    image: {
-      url: "/images/featured_post.jpg",
-      alt: "Featured Post",
-    },
+export async function processPageNews(page, api) {
+  const { blocks } = page;
+  const { docs } = await api.getCollection("news", {
+    where: { _status: { equals: "published" } },
+  });
+
+  const processArticle = (data) => ({
+    ...data,
+    author: data?.authors?.map(({ fullName }) => fullName).join(", ") ?? null,
+    image: data?.coverImage || data?.image || null,
+    date: new Date(data?.publishedOn).toUTCString(),
     link: {
-      href: "/knowledge/news",
+      href: `/knowledge/news/${data?.id}`,
     },
   });
-  blocks.push({
+
+  const articles = docs?.map(processArticle);
+
+  const featuredNewsPost = processArticle(
+    blocks.find(({ slug }) => slug === "featured-post")?.featuredPost?.value ??
+      null
+  );
+
+  const featuredPost = {
+    category: "News",
+    ...featuredNewsPost,
+    slug: "featured-post",
+  };
```

Comment:
Why don't we check if we have data first before processing it? This code looks like it will create an empty `featuredPost` (with category and slug) even if the original page data didn't have `featured-post` block.